packages/cadl-ranch-specs/http/payload/json-merge-patch/main.tsp

```diff
@@ -0,0 +1,219 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test for merge-patch+json content-type")
+@scenarioService("/json-merge-patch")
+namespace Payload.JsonMergePatch;
+
+@doc("Details about a resource.")
+model Resource {
```

Comment:
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#rest-response-body-is-resource-schema

The PATCH request schema should contain all the same fields with no required fields

Reply:
Yes. This is Azure guideline, I think in Azure world, we may recommend customer to use the core template.

Reply:
> I think in Azure world, we may recommend customer to use the core template [ResourceCreateOrUpdate](https://github.com/Azure/typespec-azure/blob/86d926a3ce78328e9bab146f15f46873cb0e1461/packages/typespec-azure-core/lib/operations.tsp#L161)

I don't think so. See https://github.com/Azure/typespec-azure-pr/issues/3688

> But in the core template, the json-merge-patch operation will have the same request and response model. -> this could be different to the guideline you pasted, so I did not add the case using Azure template before there is agreement on this.

What I pasted is restapi guideline.  No matter how you define your template, the payload should follow it. To be more specific, the raw operation you defined should follow it.

Reply:
`loadtesting` is already using required property to do patch now, `components: Record<AppComponent>;` -> `components` is required.

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/loadtestservice/LoadTestService/models.tsp#L418-L431

Reply:
If it's round trip model, I think both create(put) and update(patch) will use the same model, which means patch operation will use the model with optional property. This may violate the guideline @pshao25 pasted: [The PATCH request schema should contain all the same fields with no required fields]https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#rest-response-body-is-resource-schema

I can add `@visibility('create')` on required properties to avoid required properties to be used on patch operation. But I think we also have many unfinalized discussions on `@visibility` decorator, and `@visibility('create')` is not supported by our languages currently. Also now `loadtesting` is already using model with required property for json-merge-patch operation. (See above example). 

**So I'm wondering should json-merge-patch operation use a model that does not contain any required property?**

Add archs and typespec crew: @srnagar  @m-nash @markcowl

Reply:
To make json-merge-patch test case less ambiguous. I created another model named `ResourcePatch` which includes all the optional properties.

Reply:
They are using template because they don't have any other choices, not because it is correct usage.

And from https://learn.microsoft.com/en-us/rest/api/loadtesting/dataplane/load-test-administration/create-or-update-app-components?view=rest-loadtesting-dataplane-2022-11-01&tabs=HTTP, they have a `AppComponentUpdate` for patch operation. 

I think we still have a lot to discuss for required properties. I suggest we remove them for now until we have an aggreement.

Reply:
Add the model `ResourcePatch` to include all the optional properties.